lock_api/src/mutex.rs

```diff
@@ -169,7 +172,7 @@ impl<R: RawMutex, T: ?Sized> Mutex<R, T> {
     #[inline]
     pub fn lock(&self) -> MutexGuard<'_, R, T> {
         self.raw.lock();
-        self.guard()
+        unsafe { self.guard() }
```

Comment:
It would be good to leave a comment on each new introduced `unsafe {` discharging all proof obligations. In this case something like:
```
// SAFETY: The lock is held, as required.
```
While it is in the same file, things could be shuffled around with refactorings and it would be good to not lose these reasonings.

Reply:
Done